sdk/search/Azure.Search/src/Models/IndexDocumentsBatch{T}.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.Json;
+using Azure.Core;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Search.Models
+{
+    // Hide the untyped IndexBatch
+    [CodeGenSchema("IndexBatch")]
+    internal partial class IndexBatch { }
+
+    /// <summary>
+    /// Contains a batch of document write actions to send to a search index
+    /// via <see cref="SearchIndexClient.IndexDocuments"/>.
+    /// </summary>
+    public partial class IndexDocumentsBatch<T> : IUtf8JsonSerializable
+    {
+        /// <summary>
+        /// The actions in the batch.
+        /// </summary>
+        public IList<IndexDocumentsAction<T>> Actions { get; } = new List<IndexDocumentsAction<T>>();
+
+        /// <summary>
+        /// Initializes a new instance of the IndexDocumentsBatch class.
+        /// </summary>
+        public IndexDocumentsBatch() { }
+
+        /// <summary>
+        /// Initializes a new instance of the IndexDocumentsBatch class.
+        /// </summary>
+        /// <param name="type">
+        /// The operation to perform on a document in an indexing batch.
+        /// </param>
+        /// <param name="documents">
+        /// The collection of documents to index.
+        /// </param>
+        internal IndexDocumentsBatch(IndexActionType type, IEnumerable<T> documents)
+        {
+            Debug.Assert(Enum.IsDefined(typeof(IndexActionType), type));
+            Debug.Assert(documents != null);
+
+            if (documents != null)
+            {
+                foreach (T doc in documents)
+                {
+                    Actions.Add(new IndexDocumentsAction<T>(type, doc));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Serialize the document batch.
+        /// </summary>
+        /// <param name="writer">The JSON writer.</param>
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            Debug.Assert(writer != null);
+            writer.WriteStartObject();
+            writer.WritePropertyName("value");
+            writer.WriteStartArray();
+            foreach (IndexDocumentsAction<T> action in Actions)
+            {
+                writer.WriteObjectValue(action);
+            }
+            writer.WriteEndArray();
+            writer.WriteEndObject();
```

Comment:
The caller does it in https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/Autorest/Utf8JsonRequestContent.cs

Reply:
Do we reset/reuse the writer? If not, who's disposing it?

Reply:
@pakrym could answer this best

Reply:
Why do we need care about disposing this writer?

Reply:
The type is `IDisposable`, so doesn't it make sense to follow it's contract? 

Dispose flushes and resets the state of the writer: https://source.dot.net/#System.Text.Json/System/Text/Json/Writer/Utf8JsonWriter.cs,333

`Utf8JsonRequestContent` seems to be IDisposable as well, so I would have expected `JsonWriter` to be disposed then.

Reply:
We flush it explicitly and never reuse the writer so I guess we're fine :)

Reply:
Alright. Sounds good. Just mentioned it to future proof things, in case `Utf8JsonWriter` starts doing meaningful stuff in the `Dispose` method/relying on that being called.

If the writer is never reused, I don't see any harm in explicitly disposing of it to make sure of that.